cdap-app-fabric/src/main/java/io/cdap/cdap/app/store/ScanApplicationsRequest.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.app.store;
+
+import io.cdap.cdap.proto.id.ApplicationId;
+import io.cdap.cdap.proto.id.NamespaceId;
+import io.cdap.cdap.spi.data.SortOrder;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import javax.annotation.Nullable;
+
+/**
+ * Defines parameters of application scan within a store
+ */
+public class ScanApplicationsRequest {
+  @Nullable
+  private final NamespaceId namespaceId;
+  @Nullable
+  private final ApplicationId restartAfter;
+  private final List<ApplicationFilter> filters;
+  private final SortOrder sortOrder;
+
+  private ScanApplicationsRequest(@Nullable NamespaceId namespaceId,
+                                  @Nullable ApplicationId restartAfter,
+                                  List<ApplicationFilter> filters,
+                                  SortOrder sortOrder) {
+    this.namespaceId = namespaceId;
+    this.restartAfter = restartAfter;
+    this.filters = filters;
+    this.sortOrder = sortOrder;
+  }
+
+  /**
+   *
+   * @return namespace to return applications for or null for all namespaces
+   */
+  @Nullable
+  public NamespaceId getNamespaceId() {
+    return namespaceId;
+  }
+
+  /**
+   *
+   * @return application id to start scan from (exclusive)
+   */
+  @Nullable
+  public ApplicationId getRestartAfter() {
+    return restartAfter;
+  }
+
+  /**
+   *
+   * @return additional filters to apply. All filters must be satisfied (and operation). For performance reasons
+   * it's better to put {@link ApplicationFilter.ArtifactIdFilter} first.
+   */
+  public List<ApplicationFilter> getFilters() {
+    return filters;
+  }
+
+  /**
+   *
+   * @return sort order of the results. Results are sorted by namespace, then application id in the Ascending
+   * or Descending order.
+   */
+  public SortOrder getSortOrder() {
+    return sortOrder;
+  }
+
+  @Override
+  public String toString() {
+    return "ScanApplicationsRequest{" +
+      "namespaceId=" + namespaceId +
+      "restartAfter=" + restartAfter +
+      "filters=" + filters +
```

Comment:
can we also add toString implementation for Filter Class ?

Reply:
Done